example/network_service.rs

```diff
@@ -0,0 +1,123 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use consensus::commons::Topics;
+use consensus::messages::Message;
+use consensus::util::pending_queue::PendingQueue;
+use kadcast::config::Config;
+use kadcast::{MessageInfo, NetworkListen, Peer};
+
+use crate::wire;
+
+pub async fn run_main_loop(
+    conf: Config,
+    inbound: PendingQueue,
+    outbound: PendingQueue,
+    agr_inbound: PendingQueue,
+    agr_outbound: PendingQueue,
+) {
+    // Initialize reader and its dispatcher
+    let mut r = Reader::default();
+    r.msg_dispatcher
+        .add(Topics::Agreement as usize, agr_inbound);
+    r.msg_dispatcher.add_default(inbound);
+
+    let peer = Peer::new(conf, r);
+
+    // Broadcast outbound messages with a priority to the messages from agreement loop
+    loop {
+        tokio::select! {
+            biased;
+            recv = agr_outbound.recv() => {
+                if let Ok(msg) = recv {
+                    broadcast(&peer, msg).await;
+                }
+            }
+            recv = outbound.recv() => {
+                if let Ok(msg) = recv {
+                    broadcast(&peer, msg).await;
+                }
+            }
+
+        }
+    }
+}
+
+async fn broadcast(peer: &Peer, msg: Message) {
+    if msg.metadata.height == 0 {
+        return;
+    }
+
+    let h = (msg.metadata.height - 1) as usize;
+    peer.broadcast(&wire::Frame::encode(msg), Some(h)).await;
```

Comment:
Refactor this to call `::broadcast` with `None` height if no metadata is present

Reply:
Done.